daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -499,6 +502,45 @@ reset args = do
       fail
         "The reset command is currently not available for the HTTP JSON API. Please use the gRPC API."
 
+data LedgerExportFlags = LedgerExportFlags
+  { parties :: [String]
+  , start :: Maybe String
+  , end :: Maybe String
+  , export :: LedgerExportType
+  }
+
+data LedgerExportType = LedgerExportScript
```

Comment:
I don’t understand why you need to duplicate everything. `daml codegen java --help` works fine. What makes `daml ledger export script` different and forces us to duplicate the CLI parsing?

Reply:
That's good pointer, thanks. I had looked at `daml ledger navigator` instead, which has the same problem. `daml codegen java|scala|...` still at least replicates the command parsing, i.e. the `java|scala|...` part. But, it doesn't replicate the flags themselves. I'll look into avoiding the duplication in a follow-up.